### PR TITLE
2 5 1 patch azure windows vm

### PR DESCRIPTION
**Pull request by cmourouvin**
#### # Date     : july 2013

---
### Context
- Cloudify : **gigaspaces-cloudify-2.5.1-ga-b4200**
- JDK   : **1.6+**
- Link  : [Branch @ Fastconnect](https://github.com/fastconnect/cloudify/tree/2_5_1_patch_azure_windowsVM)
- Contact : cedric.mourouvin@fastconnect.fr
### Important
- Bootstrap you manager from a windows machine
- Run cloudify.bat as **Administrator**
- Prepare you windows image :
  - based on **Windows Server 2008 R2** template from MS Azure
  - enable 5985 port in windows firewall (_Winrm_ default communcation port)
  - create 5985 endpoint from driver configuration at **"azure.endpoints"**
- Firewall ports opening is now managed from azure-cloud.groovy at **"azure.firewall.ports"**
  - configuration [here](https://github.com/fastconnect/cloudify/tree/2_5_1_patch_azure_windowsVM/esc/src/main/resources/clouds/azure_win)
  - disabling local firewall in bootstrap-management.ps1 is desactivated (you've to open needed ports in **"azure.firewall.ports"**)

### CODE REVIEW

**Comment by barakm** [comment]
Thanks for this.
Please update the pull request to the current master (the version in master is currently 2.7.0.SNAPSHOT)

**Comment by cmourouvin** [comment]
Hello.
I'm going to do it asap.
Regards,

**Comment by gkorland** [comment]
@cmourouvin do you have an updated version for 2.7.0?

**Comment by gkorland** [comment]
@cmourouvin I'm trying to merge the changes to Cloudify 2.7.0 but I still get

```
bootstrap-client.ps1 cannot be loaded because running scripts is disabled on this system. 
For more information, seeabout_Execution_Policies at http://go.microsoft.com/fwlink/?LinkID=135170
```

**Comment by barakm** [comment]
Looks like you did not execute the client side preparations. Have a look at:
http://www.cloudifysource.org/guide/2.7/bootstrapping/bootstrapping_process.html

Look for "Operating System Specific Requirements"
You are probably missing this command:
Set-ExecutionPolicy unrestricted

File: esc/src/main/java/org/cloudifysource/esc/driver/provisioning/azure/client/MicrosoftAzureRequestBodyBuilder.java [hunk 1]
```diff
@@ -235,19 +244,45 @@ public Deployment buildDeployment(
...
+			configurationSets.getConfigurationSets().add(windowsProvisioningSet);
+			
+			// Set WinRM : HTTP without Certificate
+			WinRm winRm = new WinRm();
+			Listeners listeners = new Listeners();
+			Listener listener = new Listener();
+			listener.setCertificateThumbprint(null); // Configure for Secure Winrm command (?)
+			listener.setType("HTTP");
+			listeners.getListeners().add(listener);
+			winRm.setListeners(listeners);
```
Comment (gkorland) [inline esc/src/main/java/org/cloudifysource/esc/driver/provisioning/azure/client/MicrosoftAzureRequestBodyBuilder.java:269]: It seems like the following is missing 

```
            windowsProvisioningSet.setWinRm(winRm);
```

**Pull request closed without merging**